File: packages/amplify_core/lib/src/types/datastore/hub/network_status_event.dart
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import 'package:amplify_core/amplify_core.dart';

class NetworkStatusEvent extends DataStoreHubEventPayload {
  NetworkStatusEvent(Map<dynamic, dynamic> serializedData)
      : active = serializedData['active'] as bool;

  final bool active;
}

File: packages/amplify_datastore/darwin/amplify_datastore/Sources/amplify_datastore/AWSDataStorePluginInternal/Sync/Events/NetworkStatusEvent.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

/// Used as HubPayload for the `NetworkStatus`
public struct NetworkStatusEvent {
    /// status of network: true if network is active
    public let active: Bool

    public init(active: Bool) {
        self.active = active
    }
}
